src/vs/workbench/contrib/chat/browser/languageModelToolsService.ts

```diff
@@ -386,6 +386,9 @@ export class LanguageModelToolsService extends Disposable implements ILanguageMo
 				strs.push(part.value);
 			} else if (part.kind === 'promptTsx') {
 				strs.push(stringifyPromptTsxPart(part));
+			} else if (part.kind === 'data') {
+				strs.push(`\n\n${localize('toolResultData', "Tool result data: {0}", part.value.mimeType)}\n\n`);
+				strs.push(renderStringAsPlaintext(part.value.data.toString()));
```

Comment:
We should just say the length or something. Encoding images etc. to utf-8 strings will be meaningless

```suggestion
				strs.push(`\n\n${localize('toolResultData', "Tool result data of type {0} (length {1}", part.value.mimeType, part.value.data.length)}\n\n`);
```